components/brave_wallet_ui/components/desktop/popup-modals/transaction_details_modal/transaction_details_modal.tsx

```diff
@@ -0,0 +1,715 @@
+  transaction: BraveWallet.TransactionInfo | SerializableTransactionInfo
+  networkAsset?: BraveWallet.BlockchainToken
+  sendToken?: BraveWallet.BlockchainToken
+  buyToken?: BraveWallet.BlockchainToken
+  sellToken?: BraveWallet.BlockchainToken
+  txNetwork?: BraveWallet.NetworkInfo
+  txTypeLocale: string
+  formattedSendCurrencyTotal: string
+  formattedSendFiatValue: string
+  defaultFiatCurrency: string
+  senderLabel: string
+  recipient: string
+  recipientLabel: string
+  approvalTargetLabel: string
+  formattedSellAmount: string
+  formattedBuyAmount: string
+  formattedBuyFiatValue: string
+  spotPriceRegistry: SpotPriceRegistry | undefined
+}
+
+export const TransactionDetailsModal = (props: Props) => {
+  const {
+    onClose,
+    transaction,
+    sendToken,
+    txNetwork,
+    txTypeLocale,
+    formattedSendCurrencyTotal,
+    formattedSendFiatValue,
+    defaultFiatCurrency,
+    networkAsset,
+    senderLabel,
+    approvalTargetLabel,
+    recipient,
+    recipientLabel,
+    buyToken,
+    sellToken,
+    formattedSellAmount,
+    formattedBuyAmount,
+    formattedBuyFiatValue,
+    spotPriceRegistry
+  } = props
+
+  // Constants
+  const txCoinType = getCoinFromTxDataUnion(transaction.txDataUnion)
+  const isEthereumTx = isEthereumTransaction(transaction)
+  const isSolanaTx = isSolanaTransaction(transaction)
+  const isFilecoinTx = isFilecoinTransaction(transaction)
+  const isSwapTx = isSwapTransaction(transaction)
+  const isSolanaSwap = isSwapTx && isSolanaTx
+
+  // Queries
+  const { data: solFeeEstimates } = useGetSolanaEstimatedFeeQuery(
+    isSolanaTx &&
+      transaction.chainId &&
+      transaction.id
+      ? {
+        chainId: transaction.chainId,
+        txId: transaction.id
+      }
+      : skipToken
+  )
+
+  // Hooks
+  const onClickViewOnBlockExplorer = useExplorer(txNetwork)
+  const [speedupTx] = useSpeedupTransactionMutation()
+  const [retryTx] = useRetryTransactionMutation()
+  const [cancelTx] = useCancelTransactionMutation()
+
+  // Methods
+  const onClickRetryTransaction = () => {
+    retryTx({
+      coinType: txCoinType,
+      chainId: transaction.chainId,
+      transactionId: transaction.id
+    })
+  }
+
+  const onClickSpeedupTransaction = () => {
+    speedupTx({
+      coinType: txCoinType,
+      chainId: transaction.chainId,
+      transactionId: transaction.id
+    })
+  }
+
+  const onClickCancelTransaction = () => {
+    cancelTx({
+      coinType: txCoinType,
+      chainId: transaction.chainId,
+      transactionId: transaction.id
+    })
+  }
+
+  // Computed
+  const gasFee =
+    txCoinType === BraveWallet.CoinType.SOL
+      ? solFeeEstimates ?? ''
+      : getTransactionGasFee(transaction)
+
+  const formattedGasFeeFiatValue =
+    networkAsset && spotPriceRegistry
+      ? computeFiatAmount({
+        spotPriceRegistry,
+        value: gasFee,
+        token: networkAsset
+      }).formatAsFiat(defaultFiatCurrency) : ''
+
+  const txStatus = transaction.txStatus
```

Comment:
nit: can avoid storing a new var by using de-structuring.

```suggestion
  const { txStatus } = transaction
```